main.go

```diff
@@ -189,8 +189,11 @@ func main() { //nolint:funlen
 	// Check if user opted for disabling DSC configuration
 	_, disableDSCConfig := os.LookupEnv("DISABLE_DSC_CONFIG")
 	if !disableDSCConfig {
```

Comment:
nitpick: the disableDSCConfig flag is already gating the creation of the DSC. Instead of a separate check on the platform inside the gate, I suggest changing the default of the flag for ODH, and allowing overrides. This way we reduce complexity and still allow users to override the flag if for some reason they want an ODH installation that creates the DSCI automatically.

Reply:
i first thought about this env variable. 
it will be easier to keep the same logic in code, and add the flag in the CSV to true.

the reason i did not go this way is the name of the variable `DISABLE_DSC_CONFIG`
in reality it is about DSCI, dont remember why did we not name it as DISABLE_DSCI_CONFIG at first......
but i guess we can do this, like it is very rare if user want to set it to "false" in order to have a auto created DSCI CR. but we do need document this option somewhere.